src/utils/get-html-from-native-paste.js

```diff
@@ -0,0 +1,49 @@
+import { findDOMNode } from 'react-dom'
+
+/**
+ * Get clipboard html data by capturing the html inserted by the browser's native paste action.
+ *
+ * To make this work, `preventDefault()` may not be called on the `onPaste` event.
+ * If pasted html can be retreived, it is added to the given `data` object, setting the `type` to `html`.
+ * As this method is asynchronous, a callback is needed to return the `data` object.
+ *
+ * Solution adapted from http://stackoverflow.com/a/6804718).
+ *
+ * @param {React.Component} component
+ * @param {Object} data
+ * @param {Function} callback
+ */
+
+function getHtmlFromNativePaste(component, data, callback) {
+  const contentNode = findDOMNode(component)
+
+  const clipboardNode = contentNode.cloneNode()
+  clipboardNode.setAttribute('class', '')
+  clipboardNode.setAttribute('style', 'position: fixed; left: -9999px')
+
+  contentNode.parentNode.insertBefore(clipboardNode, contentNode)
```

Comment:
It could be done with `appendChild`as well. The original element is not removed, the cloned element is just a safe copy that receives the native paste action so the original element is left untouched.